apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js

```diff
@@ -34,4 +116,48 @@ describe('RubricsContainerTest', () => {
     const afterAddLearningGoalItems = wrapper.find('LearningGoalItem').length;
     expect(afterAddLearningGoalItems).to.equal(initialLearningGoalItems + 1);
   });
+
+  it('changes the selected level for assessment when the dropdown is changed', () => {
+    const wrapper = shallow(<RubricsContainer {...defaultProps} />);
+    let dropdown = wrapper.find('select#rubric_level_id');
+    const dropdownValue = dropdown.prop('value');
+    expect(dropdownValue).to.equal(defaultProps.levels[0].id);
+
+    dropdown.simulate('change', {target: {value: defaultProps.levels[1].id}});
+    dropdown = wrapper.find('select#rubric_level_id');
+    expect(dropdown.prop('value')).to.equal(defaultProps.levels[1].id);
+  });
+
+  it('changes the saveNotificationText and disables the save Button when saving rubric', async () => {
+    const mockFetch = sinon.stub(global, 'fetch');
+    mockFetch.returns(
+      Promise.resolve(new Response(JSON.stringify({redirectUrl: 'test_url'})))
+    );
+
+    const props = {...defaultProps, rubric: rubricInfo};
+
+    const wrapper = mount(<RubricsContainer {...props} />);
+    const notification = wrapper.find('BodyThreeText');
+
+    expect(notification.text()).not.to.contain('Saving...');
+    expect(notification.text()).not.to.contain('Save complete!');
+
+    // Simulate the save button click
+    let saveButton = wrapper.find('Button.ui-test-save-button');
+    expect(saveButton.props().disabled).to.be.false;
+    saveButton.simulate('click');
+    expect(notification.text()).to.contain('Saving...');
+    saveButton = wrapper.find('Button.ui-test-save-button');
+    expect(saveButton.props().disabled).to.be.true;
+
+    await act(async () => {
+      await Promise.resolve();
+    });
```

Comment:
I'd like to be open to new ways of testing things, but after playing with this a bit I am still not sure I understand how this is working. this makes me want to go with a better understood option like `sinon.fakeServer`.

Reply:
Thanks for the lead on `fakeServer`!  I will take a look.  The somewhat nested `awaits` were needed to get the test to pass given we are dealing with one event (clicking the save button) triggering 2 different changes on the same component (the text changing from `Saving...` to `saved!`).  It seemed like this was the only way I could figure out for slowing it down to capture both of those items.   

It always makes me uneasy when using something that isn't widely used in our codebase, so the (re)search continues onto `sinon.fakeServer`!

Reply:
thanks for the explanation kaitie, I am still not sure I get it because based on your other assertions in the code, the transition to "saving..." seems to be working before you get to the `act` call. `act` does seem like it should be fair game in the future. the two scenarios I'm seeing are:
1. to render a component (this is all of the existing examples in our codebase)
2. to do something like `act(saveButton.simulate('click'))` (apparently not needed here, but that's the idea)

Reply:
I'm suspicious that the `act` call would do anything. I think it coincidentally works to delay the next calls until after the network request logic finishes. I think it might be equivalent to just one or more `await Promise.resolve();` lines, which are basically a type of "yield" to the asynchronous threads?

The ideal thing to do would be to control the network mock in the test so you can catch the request and then have it respond predictably. Which I believe the `sinon.fakeServer` [gives you](https://sinonjs.org/releases/latest/fake-xhr-and-server/).

There, you can spy the request being made... your component code is waiting for the response. Your test then has control over that... so it can call the `respond` method of the network server mock which should synchronously move the component along. I think the test can be written, then, synchronously (without async).

Pseudo-ish-code:

```
server = sinon.fakeServer.create();
server.respondWith("GET", "/some/save/url", [200,
  { "Content-Type": "application/json" },
  '[{ "id": 1024 }]',
]);
button.simulate('click');
server.respond();

// Make sure the button talked to the network
assert(server.requests.length > 0);

// check that the button says "Saved!", etc

// Clean up
server.restore();
```

I would check conventions in other places using `fakeServer` to know what to call it and how to set it up. Probably in a `beforeEach` block.

Reply:
To somewhat tie up some lose ends, @wilkie was correct that the `act` call was essentially acting as **two** `await Promise.resolve();`.  The best I can tell, this seems to be needed in order for the `Fetch` to be resolved and then for the component to re-render with `Saved!`as the text.  I added a comment in the test to note that.